src/AstTransforms.cpp

```diff
@@ -347,14 +346,18 @@ bool MaterializeAggregationQueriesTransformer::materializeAggregationQueries(
                     visitDepthFirst(arg, [&](const AstVariable& var) { varCtr[var.getName()]++; });
                 }
             });
-            for (size_t i = 0; i < aggAtom->getArity(); i++) {
-                if (auto* var = dynamic_cast<AstVariable*>(aggAtom->getArgument(i))) {
+            std::vector<std::unique_ptr<AstArgument>> args;
+            for (auto arg : head->getArguments()) {
+                if (auto* var = dynamic_cast<AstVariable*>(arg)) {
                     // replace local variable by underscore if local
                     if (varCtr[var->getName()] == 0) {
-                        aggAtom->setArgument(i, std::make_unique<AstUnnamedVariable>());
+                        args.emplace_back(new AstUnnamedVariable());
+                        continue;
                     }
                 }
+                args.emplace_back(arg->clone());
             }
+            AstAtom* aggAtom = new AstAtom(head->getName(), std::move(args), head->getSrcLoc());
```

Comment:
Avoid repeating the type
```suggestion
            auto* aggAtom = new AstAtom(head->getName(), std::move(args), head->getSrcLoc());
```